session/txn.go

```diff
@@ -316,6 +355,22 @@ func keyNeedToLock(k, v []byte, flags tikvstore.KeyFlags) bool {
 	return !isNonUniqueIndex
 }
 
+// Info dump the TxnState to Datum for displaying in `TIDB_TRX`
+// This function is supposed to be thread safe
+func (txn *LazyTxn) Info() *txninfo.TxnInfo {
+	startTs := atomic.LoadUint64(&txn.infoStartTS)
+	if startTs == 0 {
+		return nil
+	}
+	return &txninfo.TxnInfo{
+		StartTS:        startTs,
+		State:          atomic.LoadInt32(&txn.State),
+		BlockStartTime: (*time.Time)(atomic.LoadPointer(&txn.blockStartTime)),
+		EntriesCount:   int64(txn.Len()),
```

Comment:
This visits `LazyTxn.kv.Transaction` without protection and will cause data race.
Already trying to fix it.